nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -124,10 +64,41 @@ sub boolIsTrue {
 }
 
 sub recordUnit {
-    my ($fn, $unit) = @_;
+    my ($action, $fn, $unit) = @_;
     write_file($fn, { append => 1 }, "$unit\n") if $action ne "dry-activate";
 }
 
+sub pathToUnitName {
+    my ($curSystemd, $path) = @_;
+    # Use current version of systemctl binary before daemon is reexeced.
+    open my $cmd, "-|", "$curSystemd/systemd-escape", "--suffix=mount", "-p", $path
+        or die "Unable to escape $path!\n";
```

Comment:
```suggestion
    open(my $cmd, "-|", "$curSystemd/systemd-escape", "--suffix=mount", "-p", $path)
        or die "Unable to escape $path! $!\n";
```

I'd consider parentheses here to avoid pitfalls with precedence with `or` and `||`, also nice to include any error message set in `$!`

Reply:
Applied in https://github.com/Mic92/nixpkgs/commit/e455170f0869021a2a0e0720ac2f29cf22dbaf84 but let's fix this later.